great_expectations/execution_engine/execution_engine.py

```diff
@@ -494,6 +494,58 @@ def resolve_data_reference(
             template_arguments=template_arguments,
         )
 
+    @staticmethod
+    def _split_table_metric_domain_kwargs(
+        domain_kwargs: Dict,
+        domain_type: Union[str, MetricDomainTypes],
+        accessor_keys: Optional[Iterable[str]] = None,
+    ) -> Tuple[Dict, Dict]:
+        """Split domain_kwargs for table domain types into compute and accessor domain kwargs.
+
+        Args:
+            domain_kwargs: A dictionary consisting of the domain kwargs specifying which data to obtain
+            domain_type: an Enum value indicating which metric domain the user would
+            like to be using, or a corresponding string value representing it. String types include "identity",
+            "column", "column_pair", "table" and "other". Enum types include capitalized versions of these from the
+            class MetricDomainTypes.
+            accessor_keys: keys that are part of the compute domain but should be ignored when
+            describing the domain and simply transferred with their associated values into accessor_domain_kwargs.
+
+        Returns:
+            compute_domain_kwargs, accessor_domain_kwargs from domain_kwargs
+            The union of compute_domain_kwargs, accessor_domain_kwargs is the input domain_kwargs
+        """
```

Comment:
really loving this refactor @anthonyburdi... again setting the bar for doc strings :)

Reply:
Thank you! But most of this is copy pasted from the parent so I can't take the credit :)